vyos/vyos-1x
# smoketest: T5558: Extend configtest to allow checking of migration script results

## Change Summary

- Extends config smoketest to check resulting migrated CLI nodes against a file of expected nodes
- Includes list of expected CLI nodes for tests `basic-vyos` and `dialup-router-medium-vpn`

Will add further configs/expected CLI for major migrated components (e.g. firewall/zone-policy) over time.

Note: As future migrations are added that change syntax, tests may need to be updated.

## Types of changes


## Related Task(s)

* https://vyos.dev/T5558

## Component(s) name

smoketest

## Proposed changes


## How to test

Run `make testc`, after migration is loaded and committed, it will check config commands in the `config-tests` file (if present) against the effective config.

## Checklist:
